sources/packages/backend/apps/api/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -236,4 +246,118 @@ export class StudentAssessmentService extends RecordDataModelService<StudentAsse
+              id: true,
+            },
+          },
+          studentAssessments: {
+            id: true,
+            studentAssessmentStatus: true,
+          },
+          student: { id: true },
+        },
+        relations: {
+          currentAssessment: { offering: true, studentAppeal: true },
+          studentAssessments: true,
+          student: true,
+        },
+        where: {
+          id: applicationId,
+          studentAssessments: {
+            triggerType: AssessmentTriggerType.OriginalAssessment,
+          },
+        },
+      });
+
+      if (!application) {
+        throw new CustomNamedError(
+          `Application not found.`,
+          APPLICATION_NOT_FOUND,
+        );
+      }
+      if (application.isArchived) {
+        throw new CustomNamedError(
+          `Application cannot have manual reassessment after being archived.`,
+          INVALID_OPERATION_IN_THE_CURRENT_STATUS,
+        );
+      }
+      const [originalAssessment] = application.studentAssessments;
+      if (
+        originalAssessment.studentAssessmentStatus !==
+        StudentAssessmentStatus.Completed
+      ) {
+        throw new CustomNamedError(
+          `Application original assessment expected to be '${StudentAssessmentStatus.Completed}' to allow manual reassessment.`,
+          INVALID_OPERATION_IN_THE_CURRENT_STATUS,
+        );
+      }
+
+      // Adds application note.
+      const newNote = new Note();
+      newNote.description = note;
+      newNote.noteType = NoteType.Application;
+      newNote.creator = currentUser;
+      newNote.createdAt = now;
+      const savedNote = await transactionalEntityManager
+        .getRepository(Note)
+        .save(newNote);
+      // Associates the created note with the student.
+      await transactionalEntityManager
+        .getRepository(Student)
+        .createQueryBuilder()
+        .relation(Student, "notes")
+        .of(application.student)
+        .add(savedNote);
+
+      const oldCurrentAssessment = application.currentAssessment;
+      const newCurrentAssessment = new StudentAssessment();
+      newCurrentAssessment.triggerType =
+        AssessmentTriggerType.ManualReassessment;
+      newCurrentAssessment.studentAssessmentStatus =
+        StudentAssessmentStatus.Submitted;
+      newCurrentAssessment.studentAssessmentStatusUpdatedOn = now;
+      newCurrentAssessment.offering = oldCurrentAssessment.offering;
+      newCurrentAssessment.studentAppeal = oldCurrentAssessment.studentAppeal;
+      newCurrentAssessment.submittedDate = now;
+      newCurrentAssessment.submittedBy = currentUser;
+      newCurrentAssessment.createdAt = now;
+      newCurrentAssessment.creator = currentUser;
+      newCurrentAssessment.application = application;
+
+      await transactionalEntityManager
+        .getRepository(StudentAssessment)
+        .insert(newCurrentAssessment);
+
+      application.currentAssessment = newCurrentAssessment;
+      await applicationRepo.update(
```

Comment:
Eventhough this code is going to change, please make sure audit fields are populated in every insert/update.